### PR TITLE
Avoid clode for the last recv of a message

### PR DESCRIPTION
**Pull request by Licenser**
This removes the clone mentioned #18 for `try_recv` if we are the last receiver reading a message that would move the queue forward.

It also adds two benchmarks (one sender, one receiver) and (one sender two receivers) to measure this. The benchmark is not perfect as it uses simple_features as well as a copy-able payload (the best possible case for the old code) the results were still encouraging.

For 1->1 it showed a 1% drop in performance
for 1->2 it showed a 9% improvement in perforamcen

that seems worth it as we'll likely see much bigger improvements for more complex data structures that need to be cloned instead of copied.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
+categories = []
+description = "Async broadcast channels"
+documentation = "https://docs.rs/async-broadcast"
+edition = "2018"
+keywords = []
+license = "MIT OR Apache-2.0"
+name = "async-broadcast"
+readme = "README.md"
+repository = "https://github.com/smol-rs/async-broadcast"
+version = "0.3.3"
```
Comment (zeenix) [inline Cargo.toml:16]: These changes are separate logical changes and hence should be in a separate commit w/ its own summary (and possibly rationale).

File: Cargo.toml [hunk 2]
```diff
@@ -1,28 +1,33 @@
...
 
 [dev-dependencies]
+criterion = "0.3"
 doc-comment = "0.3.3"
 futures-lite = "1.11.3"
 futures-util = "0.3.14"
+
+[[bench]]
+harness = false
+name = "broadcast_bench"
```
Comment (zeenix) [inline Cargo.toml:33]: addition of benchmarking is awesome! However it's also a separate logical change so separate commit please.

**Comment by zeenix** [comment]
@Licenser Thanks so much for this great contribution! :+1: The actual changes look good to me, so I only have nitpicks about non-functional aspects. :) Apart from the comments on the commit:

* Please also put the rationale/description on the (relevant) commit log description as well.
* Typo in the description: " improvement in perforamcen".

Otherwise, looks really great.

**Comment by Licenser** [comment]
:+1: split it out into two commits and added some additional text to them

File: Cargo.toml (hunk 1 above)
Comment (zeenix) [reply Cargo.toml:16]: This part is still not relevant to the commit it's in.

**Comment by zeenix** [comment]
> +1 split it out into two commits and added some additional text to them

Awesome! That was quick. Just one more split and we're good to go. :)

**Comment by Licenser** [comment]
Ohh, sorry I didn't understand what you meant, the changes on the `Cargo.toml` were VSCode being eager :) I reverted them as I didn't mean to change the `Cargo.toml` aside from the addition of benchmarks and the new dependency. Hope this looks good now

**Comment by zeenix** [comment]
Many thanks @Licenser !
